projects/packages/forms/src/blocks/shared/util/block-utils.js

```diff
@@ -0,0 +1,36 @@
+import { select, useSelect } from '@wordpress/data';
+
+/**
+ * Determines if the current block selection is within a contact form.
+ *
+ * @return {boolean} Whether the selection is within a contact form
+ */
+export const isWithinContactForm = () => {
+	const blockEditor = select( 'core/block-editor' );
+	if ( ! blockEditor ) {
+		return false;
+	}
+
+	const selectedBlockClientIds = blockEditor.getSelectedBlockClientIds();
+	if ( ! selectedBlockClientIds?.length ) {
+		return false;
+	}
+
+	return selectedBlockClientIds.some( blockId => {
+		const parentBlocks = blockEditor.getBlockParentsByBlockName( blockId, 'jetpack/contact-form' );
+		return parentBlocks.length > 0;
+	} );
+};
+
+/**
+ * React hook wrapper around `isWithinContactForm` that re-evaluates whenever the
+ * current selection changes.
+ *
+ * Components should prefer this over calling the util directly so that they
+ * remain reactive without manual subscriptions.
+ *
+ * @return {boolean} Whether the selection is within a contact form
+ *
+ * Usage:
+ */
+export const useIsWithinContactForm = () => useSelect( () => isWithinContactForm(), [] );
```

Comment:
```suggestion
export const useIsWithinContactForm = () => useSelect( ( select ) => isWithinContactForm( select ), [] );
```
I think you should consider passing `select` into `isWithinContactForm` from `useSelect`. `useSelect` and its `select` argument guarantees you're the correct store registry (which is passed via react context, the same as `useRegistry`), so it's preferred.